INSTALL.md

```diff
@@ -0,0 +1,551 @@
+in the `result/bin` directory.
+
+</details>
+
+<details>
+
+<summary>Setup a development environment with all required dependencies and SMT solvers</summary>
+  To setup a development environment, run the following command:
+
+  ```
+  make nix-develop
+  ```
+
+  This installs all required dependencies and SMT solvers, and drops you in a
+  shell afterward. From there, run `make` to build EasyCrypt.
+
+</details>
+
+## From Source
+
+> **Note:** As this is not the recommended installation method and is
+> considered more advanced, we don't go into much detail here and primarily
+> refer to external sources. If you are looking for a more guided installation
+> process, consider [installing via OPAM](#via-opam-recommended) instead.
+
+### Installing EasyCrypt's Dependencies Manually
+
+> **Note:** If you only want to install EasyCrypt *itself* from source but prefer
+> to install *its dependencies* automatically, consider following the
+> [OPAM-based installation instructions](#via-opam-recommended) up to and including
+> [the installation of EasyCrypt's dependencies](installing-easycrypts-dependencies-via-opam),
+> then proceed to [install Easycrypt from source](#installing-easycrypt-from-source).
+
+The following is a list of all the dependencies required to build and run
+EasyCrypt. For each dependency, we provide some installation resources along
+with any relevant details or caveats. Once all dependencies are installed,
+proceed to [install EasyCrypt from Source](#installing-easycrypt-from-source).
+
+- [OCaml](https://ocaml.org/install/) (version >= 4.08 and < 5.0)  
+  Additional resources:
+  - <https://github.com/ocaml/ocaml>.
+- [OCamlbuild](https://github.com/ocaml/ocamlbuild/)
+- [Git](https://git-scm.com/downloads)
+- [Why3](https://www.why3.org/) (version >= 1.8 and < 1.9)  
+  You also need to install Why3's libraries (typically by running `make byte && make install-lib`).
+  Further, you might already want to install at least one [compatible SMT solver](README.md#compatibility).  
+  Additional resources:
+  - <https://www.why3.org/#provers>
+  - <https://gitlab.inria.fr/why3/why3>
+- [Menhir](https://gallium.inria.fr/~fpottier/menhir/)  
+  Additional resources:
+  - <https://gitlab.inria.fr/fpottier/menhir>
+- [OCaml Batteries Included](https://ocaml-batteries-team.github.io/batteries-included/hdoc2/) (version >= 3)  
+  Additional resources:
+  - <https://github.com/ocaml-batteries-team/batteries-included>
+  - <http://batteries.forge.ocamlcore.org>
+- [OCaml PCRE](https://github.com/mmottl/pcre-ocaml/) (version >= 7)
+- [OCaml Zarith](https://github.com/ocaml/Zarith/) (version >= 1.10)  
+  Additional resources:
+  - <https://forge.ocamlcore.org/projects/zarith>
+- [OCaml ini-files](https://opam.ocaml.org/packages/ocaml-inifiles/) (version >= 1.2)  
+  Additional resources:
+  - http://archive.ubuntu.com/ubuntu/pool/universe/o/ocaml-inifiles
+- [Python3](https://www.python.org/downloads)
```

Comment:
Still missing python-curses?
https://docs.python.org/3/howto/curses.html#the-python-curses-module

Reply:
Isn't this a built-in module?

Reply:
I had to set it up separately on OpenSUSE tumbleweed. But in Python's defence, it gives an error message which is at least as good as what EasyCrypt gives when you try to require a theory that doesn't exist. (Read: it's shit and gives no hint about how to fix it to a user who's not familiar with python.)